### PR TITLE
Bump golangci version to v1.38.0

### LINKED ISSUE #17149: Update golangci to v1.39.0
The goal of this ticket is to update golangci-lint to v1.38.0 and fix all issues reported by the `make check-style` command.

----
If you're interested please comment here and come [join our "Contributors" community channel](https://community.mattermost.com/core/channels/tickets) on our daily build server, where you can discuss questions with community members and the Mattermost core team. For technical advice or questions, please  [join our "Developers" community channel](https://community.mattermost.com/core/channels/developers).

New contributors please see our [Developer's Guide](https://developers.mattermost.com/contribute/getting-started/).

JIRA: https://mattermost.atlassian.net/browse/MM-33870

### PR DESCRIPTION
**Pull request by haardikdharma10**
#### Summary

For #17149 

#### Ticket Link

Fixes #17149

#### Release Note

```release-note
NONE
```

### CODE REVIEW

**Comment by streamer45** [comment]
@haardikdharma10 

Thanks for the PR. Now that the tool is updated we should fix the reported errors. You should be able to reproduce them locally by running `make check-style`. Just make sure your local `golanci-lint` version has been updated aswell.

```sh
$ golangci-lint version          
golangci-lint has version v1.38.0
```

**Comment by agnivade** [comment]
Please also revert any changes go.mod and go.sum. They should not be modified.

**Comment by streamer45** [comment]
@haardikdharma10 . Are you still planning on working on this? If not, we can let someone else take a look. Let us know, thanks :)

**Comment by haardikdharma10** [comment]
These are three linter errors that still persist. Can you help me out here? @streamer45
```
api4/terms_of_service_test.go:46:2: ineffectual assignment to termsOfService (ineffassign)
        termsOfService, resp := Client.CreateTermsOfService("terms of service new", th.SystemAdminUser.Id)
        ^
model/integration_action_test.go:78:17: ineffectual assignment to err (ineffassign)
                _, triggerId, err := GenerateTriggerId(NewId(), key)
                              ^
app/slashcommands/command_test.go:174:2: ineffectual assignment to builtIn (ineffassign)
        builtIn = false
```

File: app/slashcommands/command_test.go [hunk 1]
```diff
@@ -171,8 +171,6 @@ func TestHandleCommandResponsePost(t *testing.T) {
 	assert.Nil(t, post.GetProp("from_webhook"))
 
 	// Command is not built in, so it is a bot command.
-	builtIn = false
-	post, err = th.App.HandleCommandResponsePost(command, args, resp, builtIn)
```
Comment (streamer45) [inline app/slashcommands/command_test.go:175]: We should leave this and add a check for the `err`.
```go
assert.Nil(t, err)
```

**Review by streamer45 — COMMENTED** [review]
@haardikdharma10 

`api4/terms_of_service_test.go`

Let's try to add a check for the returned value:

```go
assert.Nil(t, termsOfService)
```

`model/integration_action_test.go`

Let's check the returned error:

```go
require.NotNil(t, err)
```

File: .circleci/config.yml [hunk 2]
```diff
@@ -163,7 +163,7 @@ jobs:
   # and to make more clear when the job fails
   check-golangci-lint:
     docker:
-      - image: golangci/golangci-lint:v1.33.2
+      - image: golangci/golangci-lint:v1.38.0
```
Comment (cpanato) [inline .circleci/config.yml:166]: why not use 1.39.0? which is the latest

Comment (streamer45) [reply .circleci/config.yml:166]: No reason, it's been a while since the PR was opened. We can update ;)

**Review by cpanato — APPROVED** [review]

File: .circleci/config.yml (hunk 2 above)
Comment (cpanato) [reply .circleci/config.yml:166]: i will do in a followup, lets get this merged

**Review by streamer45 — APPROVED** [review]

**Pull request merged**